File: java/quarkus/mongo-client/src/main/java/org/acme/data/Book.java
package org.acme.data;

public class Book {
    private String title;
    private Integer pages;

    public Book(String title, Integer pages) {
        this.title = title;
        this.pages = pages;
    }

    public String getTitle() {
        return title;
    }

    public Integer getPages() {
        return pages;
    }
}

File: java/quarkus/datasource-hibernate-search-elasticsearch/src/main/kotlin/org/acme/Book.kt
package org.acme

import com.fasterxml.jackson.annotation.JsonIgnore
import io.quarkus.hibernate.orm.panache.kotlin.PanacheCompanion
import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntity
import org.hibernate.search.mapper.pojo.mapping.definition.annotation.FullTextField
import org.hibernate.search.mapper.pojo.mapping.definition.annotation.Indexed
import javax.persistence.Entity
import javax.persistence.ManyToOne
import javax.persistence.Table

@Entity
@Indexed
@Table(name = "book", schema = "hello")
data class Book(
        @FullTextField(analyzer = "english")
        var title: String,

        @field: JsonIgnore
        @field: ManyToOne
        var author: Author,

        var pages: Int
) : PanacheEntity() {
    companion object : PanacheCompanion<Book, Long> {
        fun findAllBooks() = findAll().list()
    }
}
